### PR TITLE
add ramp_leaderboard command line tool

### PR DESCRIPTION
**Pull request by mehdidc**
Add a simple command line "ramp_leaderboard" (https://github.com/mehdidc/ramp-leaderboard) to display a leaderboard of local submissions.
(see #132)

### CODE REVIEW

File: rampwf/utils/command_line.py [hunk 1]
```diff
@@ -166,3 +171,270 @@ def ramp_blend_submissions():
...
+        metric = metrics[0]
+        sort_cols = ['{}_{}'.format(step, metric)
+                     for step in ('test', 'valid', 'train')]
+    df = df.sort_values(by=sort_cols, ascending=args.asc)
+    df = df[['submission'] + show_cols]
+    print(tabulate(df, headers='keys', tablefmt='grid'))
+
+
+def _get_metrics(df):
+    metrics = [
```
Comment (aboucaud) [inline rampwf/utils/command_line.py]: Add a short docstring

File: rampwf/utils/command_line.py [hunk 2]
```diff
@@ -166,3 +171,270 @@ def ramp_blend_submissions():
...
+        c.split('_')[1]
+        for c in df.columns if c.startswith('train')
+    ]
+    metrics = set(metrics)
+    metrics = list(metrics)
+    return metrics
+
+
+def _build_leaderboard_df(scores_dict, precision=2):
+    rows = []
```
Comment (aboucaud) [inline rampwf/utils/command_line.py]: same here

File: rampwf/utils/command_line.py [hunk 3]
```diff
@@ -166,3 +171,270 @@ def ramp_blend_submissions():
...
+        '{}_{}_{}'.format(step, metric, part)
+        for metric in mean_scores.columns
+        for step in ('train', 'valid', 'test')
+        for part in ('mean', 'std')
+    ]
+    return pd.DataFrame(rows, columns=['submission'] + columns_order)
+
+
+def _build_scores_dict(ramp_kit_dir='.'):
+    submissions_folder = os.path.join(ramp_kit_dir, 'submissions')
```
Comment (aboucaud) [inline rampwf/utils/command_line.py]: same here

**Comment by mehdidc** [comment]
Or we could put in its own  repo in paris-saclay-cds, as it is not necessary to ramp-workflow

**Comment by kegl** [comment]
I have this when running it on the titanic starting kit.

```silver6:titanic kegl$ ramp_leaderboard 
Traceback (most recent call last):
  File "/Users/kegl/anaconda/bin/ramp_leaderboard", line 11, in <module>
    load_entry_point('ramp-workflow', 'console_scripts', 'ramp_leaderboard')()
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 480, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2693, in load_entry_point
    return ep.load()
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2324, in load
    return self.resolve()
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2330, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/kegl/Research/RAMP/ramp-workflow/rampwf/__init__.py", line 4, in <module>
    from . import utils
  File "/Users/kegl/Research/RAMP/ramp-workflow/rampwf/utils/__init__.py", line 1, in <module>
    from .command_line import (
  File "/Users/kegl/Research/RAMP/ramp-workflow/rampwf/utils/command_line.py", line 402
SyntaxError: Non-ASCII character '\xc2' in file /Users/kegl/Research/RAMP/ramp-workflow/rampwf/utils/command_line.py on line 402, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

**Comment by agramfort** [comment]
any chance to have tests?

let's try to avoid merging code without tests with good coverage...

**Comment by glemaitre** [comment]
I would vendor `tabulate` if this is a single file and that we are not attracted by new feature.
I would probably do the same for the `colored` package.

**Comment by glemaitre** [comment]
By vendoring I mean to create an `externals` module in `rampwf`

**Pull request closed without merging**